packages/manager-react-components/src/hooks/datagrid/useIcebergV6.tsx

```diff
@@ -24,73 +19,48 @@ export function useResourcesIcebergV6<T = unknown>({
   queryKey,
   defaultSorting = undefined,
 }: IcebergFetchParamsV6 & IcebergV6Hook) {
-  const [totalCount, setTotalCount] = useState(0);
-  const [flattenData, setFlattenData] = useState<T[]>([]);
-  const [queryParams, setQueryParams] = useState<QueryParams>({
-    pageIndex: 1,
-    sorting: { desc: false, id: '' },
-  });
   const [sorting, setSorting] = useState<ColumnSort>(defaultSorting);
 
   const {
-    data,
+    data: dataSelected,
     fetchNextPage,
     isError,
     isLoading,
     error,
     status,
   } = useInfiniteQuery({
-    initialPageParam: null,
+    initialPageParam: 1,
     queryKey: [...queryKey, sorting],
-    queryFn: ({ pageParam }) =>
+    staleTime: Infinity,
+    retry: false,
```

Comment:
These properties should be given through parameters of the hook, to let the caller use it at its convenience and to prevent from being too specific for a general purpose.

Reply:
I approve. 

But It's a Breaking change. 

And I don't want to do it on this PR. The first priority is fix bug. And I take this opportunity for do a refacto.

But I avoid create bug or blocked this PR for that.

May be another ticket ?

Reply:
Yep, agree 👍🏻

Reply:
I will add that we need a deep refactor of this hook, because, as it is, it's not generic at all, I mean a lot of parameters are hard coded and the `select` function itself should be a hook parameter so the caller could do whatever he wants with the raw data.